test/state_test.js

```diff
@@ -217,24 +234,36 @@ describe('State', () => {
     assume(spotRequests).has.length(1);
     assume(spotRequests[0]).has.property('state', secondState);
   });
+   
+  it('should be able to update an AMI\'s usage', async () => {
+    await db.insertAmiUsage({region: defaultSR.region, id: defaultSR.id});
+    let amiUsage = await db.listAmiUsage(); 
+    assume(amiUsage).has.length(1);
+    
+    await db.updateAmiUsage({
+      region: 'us-west-1',
+      id: defaultSR.id,
+    });
+     
+    amiUsage = await db.listAmiUsage();
+    assume(amiUsage).has.length(1);
+    assume(amiUsage[0]).has.property('region', 'us-west-1');
+  });
 
-  it('should be able to do a spot request upsert', async () => {
-    let firstState = 'open';
-    let secondState = 'closed';
-    defaultSR.state = firstState;
+  it('should be able to upsert an AMI\'s usage', async () => {
 
-    await db.upsertSpotRequest(defaultSR);
-    let spotRequests = await db.listSpotRequests(); 
-    assume(spotRequests).has.length(1);
-    assume(spotRequests[0]).has.property('state', firstState);
+    await db.upsertAmiUsage({region: defaultSR.region, id: defaultSR.id});
+    let amiUsage = await db.listAmiUsage(); 
+    assume(amiUsage).has.length(1);
+    assume(amiUsage[0]).has.property('region', defaultSR.region);
 
-    defaultSR.state = secondState;
-    await db.upsertSpotRequest(defaultSR);
-    spotRequests = await db.listSpotRequests(); 
-    assume(spotRequests).has.length(1);
-    assume(spotRequests[0]).has.property('state', secondState);
+    let newRegion = 'us-west-1';
+    await db.upsertAmiUsage({region: newRegion, id: defaultSR.id});
```

Comment:
This test isn't really quite on the mark.  We're defining the a `newRegion` variable, which is actually the same string as `defaultSR.region`.  If those were different strings, then this test would fail because it wouldn't result in a primary key collision and would insert a second row.